flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeScheduler.java

```diff
@@ -725,27 +729,32 @@ public boolean canScaleUp(ExecutionGraph executionGraph) {
         int availableSlots = declarativeSlotPool.getFreeSlotsInformation().size();
 
         if (availableSlots > 0) {
-            // check if rescaling would increase the parallelism of at least one vertex
-
             final Optional<? extends VertexAssignment> potentialNewParallelism =
                     slotAllocator.determineParallelism(
                             jobInformation, declarativeSlotPool.getAllSlotsInformation());
 
-            return potentialNewParallelism.isPresent()
-                    && isParallelismHigher(potentialNewParallelism.get(), executionGraph);
+            if (potentialNewParallelism.isPresent()) {
+                final VertexAssignment currentParallelism =
+                        slotAllocator.determineParallelism(executionGraph);
+                int currentCumulativeParallelism = getCumulativeParallelism(currentParallelism);
+                int newCumulativeParallelism =
+                        getCumulativeParallelism(potentialNewParallelism.get());
+                if (newCumulativeParallelism > currentCumulativeParallelism) {
```

Comment:
Do we always want to scale up if we can? What if we are already running with our desired parallelism? Should the `ScalingPolicy` decide this or should the scheduler already make the decision?

Reply:
I think this cannot happen because `slotAllocator.determineParallelism` should not determine a different parallelism if we are already running at our desired parallelism.